setup.py

```diff
@@ -6,73 +6,74 @@
-        "pydocstyle>=3.0.0,<4",
+    "lint": [
+        "flake8==6.0.0",  # flake8 claims semver but adds new warnings at minor releases, leave it pinned.
+        "flake8-bugbear==23.3.23",  # flake8-bugbear does not follow semver, leave it pinned.
+        "isort>=5.10.1",
+        "mypy==0.971",  # mypy does not follow semver, leave it pinned.
+        "pydocstyle>=6.0.0",
+        "black>=22",
     ],
-    'doc': [
+    "doc": [
         "sphinx>=5.0.0",
         "sphinx_rtd_theme>=1.0.0",
         "towncrier>=21,<22",
     ],
-    'dev': [
-        "bumpversion>=0.5.3,<1",
-        "pytest-watch>=4.1.0,<5",
+    "dev": [
+        "bumpversion>=0.5.3",
+        "pytest-watch>=4.1.0",
+        "tox>=4.0.0",
         "wheel",
         "twine",
         "ipython",
     ],
 }
 
-extras_require['dev'] = (
-    extras_require['dev'] +  # noqa: W504
-    extras_require['test'] +  # noqa: W504
-    extras_require['lint'] +  # noqa: W504
-    extras_require['doc']
+extras_require["dev"] = (
+    extras_require["dev"]
+    + extras_require["test"]
+    + extras_require["lint"]
+    + extras_require["doc"]
 )
 
 
-with open('./README.md') as readme:
+with open("./README.md") as readme:
     long_description = readme.read()
 
 
 setup(
-    name='eth-typing',
+    name="eth-typing",
     # *IMPORTANT*: Don't manually change the version here. Use `make bump`, as described in readme
-    version='3.3.0',
+    version="3.3.0",
     description="""eth-typing: Common type annotations for ethereum python packages""",
     long_description=long_description,
-    long_description_content_type='text/markdown',
-    author='The Ethereum Foundation',
-    author_email='snakecharmers@ethereum.org',
-    url='https://github.com/ethereum/eth-typing',
+    long_description_content_type="text/markdown",
+    author="The Ethereum Foundation",
+    author_email="snakecharmers@ethereum.org",
+    url="https://github.com/ethereum/eth-typing",
     include_package_data=True,
-    python_requires='>=3.7.2, <4',
+    python_requires=">=3.7, <4",
     extras_require=extras_require,
-    py_modules=['eth_typing'],
+    py_modules=["eth_typing"],
     license="MIT",
     zip_safe=False,
-    keywords='ethereum',
+    keywords="ethereum",
     packages=find_packages(exclude=["tests", "tests.*"]),
-    package_data={'eth_typing': ['py.typed']},
+    package_data={"eth_typing": ["py.typed"]},
     classifiers=[
-        'Development Status :: 5 - Production/Stable',
-        'Intended Audience :: Developers',
-        'License :: OSI Approved :: MIT License',
-        'Natural Language :: English',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.7',
-        'Programming Language :: Python :: 3.8',
-        'Programming Language :: Python :: 3.9',
-        'Programming Language :: Python :: 3.10',
-        'Programming Language :: Python :: 3.11',
-        'Programming Language :: Python :: Implementation :: PyPy',
```

Comment:
I think this should still work with `pypy`?

Reply:
We had talked last week(ish) about removing PyPy builds from CI, so I removed the reference here too. Should I add it back if we're not testing it?

Reply:
Nah, don't mind me :)